app/models/auth/voms.rb

```diff
@@ -41,11 +41,21 @@ def parse_hash_groups!(hash)
         raise Errors::AuthenticationError, 'voms group env variable is not set' unless hash.key?('GRST_VOMS_FQANS')
         groups = Hash.new { |h, k| h[k] = [] }
         hash['GRST_VOMS_FQANS'].split(';').each do |line|
-          matches = line.match(VOMS_GROUP_REGEXP)
-          groups[matches[:group]] << matches[:role] if matches && matches[:role] != 'NULL'
+          group = parse_group(line)
+          groups.merge!(group) { |_, oldval, newval| oldval + newval } if group
```

Comment:
Should we run the resulting array(s) with roles through `uniq!` and possibly `sort!` at the end? It could improve traceability and prevent various "chicken or egg" problems with roles later on. Or shoud that be part of the unification process?

Reply:
I'll write a few tests to see if such problems can appear. If something will show up i'll reply.

Reply:
You were right.
expected: [{:id=>"coolclub", :roles=>["gamers"]}]
         got: [{:id=>"coolclub", :roles=>["gamers", "gamers"]}]
So I think that we should use uniq! at least.